api/app/Repositories/UserRepository.php

```diff
@@ -76,4 +86,36 @@ public function makeLoginToken($id)
 
         return $token;
     }
+
+    /**
+     * Make an email confirmation token for a user.
+     *
+     * @param string $email
+     *
+     * @return string
+     */
+    public function makeConfirmationToken($email)
+    {
+        $token = hash_hmac('sha256', str_random(40), str_random(40));
+        $this->cache->put('email_confirmation_'.$token, $email, $this->confirmation_token_ttl);
+        return $token;
+    }
+
+    /**
+     * If the email_confirmation field is set, make sure we've a valid token.
+     *
+     * @param  Request  $request
+     * @return void
```

Comment:
As this method validates something, it should return true so that it can be wrapped in an `if(validateEmailConfirmationToken($token))` call.